webdriver/tests/interaction/element_clear.py

```diff
@@ -0,0 +1,133 @@
+    assert_error(response, "invalid element state")
+
+
+def test_scroll_into_element_view(session):
+    # 14.2 Step 4
+    session.url = inline("<input type=text value=Federer><div style= \"height: 200vh; width: 5000vh\">")
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = clear(session, element)
+
+    # Check if element cleared is scrolled into view
+    rect = session.execute_script("return document.getElementsByTagName(\"input\")[0].getBoundingClientRect()")
+
+    pageDict = {}
+
+    pageDict["innerHeight"] = session.execute_script("return window.innerHeight")
+    pageDict["innerWidth"] = session.execute_script("return window.innerWidth")
+    pageDict["pageXOffset"] = session.execute_script("return window.pageXOffset")
+    pageDict["pageYOffset"] = session.execute_script("return window.pageYOffset")
+
+    assert rect["top"] < (pageDict["innerHeight"] + pageDict["pageYOffset"]) and \
+           rect["left"] < (pageDict["innerWidth"] + pageDict["pageXOffset"]) and \
+           (rect["top"] + element.rect["height"]) > pageDict["pageYOffset"] and \
+           (rect["left"] + element.rect["width"]) > pageDict["pageXOffset"]
+
+
+# TODO
+# Any suggestions on implementation?
+# def test_session_implicit_wait_timeout(session):
+    # 14.2 Step 5
+
+# TODO
+# Any suggestions on implementation?
+# def test_element_not_interactable(session):
+#     # 14.2 Step 6
+#     assert_error(response, "element not interactable")
+
+
+def test_element_readonly(session):
+    # 14.2 Step 7
+    session.url = inline("<input type=text readonly value=Federer>")
+
+    element = session.find.css("input", all=False)
+    response = clear(session, element)
+    assert_error(response, "invalid element state")
+
+
+def test_element_disabled(session):
+    # 14.2 Step 7
+    session.url = inline("<input type=text disabled value=Federer>")
+
+    element = session.find.css("input", all=False)
+    response = clear(session, element)
+    assert_error(response, "invalid element state")
+
+
+def test_element_pointer_events_disabled(session):
+    # 14.2 Step 7
+    session.url = inline("<input type=text value=Federer style=\"pointer-events: none\">")
+
+    element = session.find.css("input", all=False)
+    response = clear(session, element)
+    assert_error(response, "invalid element state")
+
+
+def test_clear_editable_element(session):
+    # 14.2 Step 8
+    session.url = inline("<p contenteditable=true>This is an editable paragraph.")
+
+    element = session.find.css("p", all=False)
+    response = clear(session, element)
+
+    assert element.text == ""
+
+
+def test_clear_resettable_element(session):
+    # 14.2 Step 8
+    session.url = inline("<input type=text value=Federer>")
+
+    element = session.find.css("input", all=False)
```

Comment:
This doesn’t test all the steps for [clear a resettable element](https://w3c.github.io/webdriver/webdriver-spec.html#dfn-clear-a-resettable-element). For example you should check with an `<input>` without a value (of different types), that is gets focussed, that the reset algorithm works for all resettable elements, and that it gets unfocussed.

Reply:
Hey @andreastt,

So I parametrized my clear resettable elements: input: (text, search, url, telephone, email, password, date, time, number, range, color, file), textarea, select and output elements. 

Right now the test fails clearing date, time, range, color, select and output elements. If the test fails one test case, it will stop the test. Should I make an individual test for each element? 

Also, none of the webdrivers sets focus and unfocus when the element is being cleared. Should I file a bug for the browsers or the webdriver specification?